### PR TITLE
Hotfix #76 - Parsing string data

### PR DESCRIPTION
**Pull request by webimpress**
Fix for #76.

I couldn't find any example of `$content` when `parse_str($content, $bodyParams);` will store in `$bodyParams` non-array value.

The removed condition, stolen from `AbstractRestfulController`, in `ContentTypeListener` was stolen incorrectly.
The original condition in `AbstractRestfulController` is (with comment, and renamed variables to fit our code):

```php
// If parse_str fails to decode, or we have a single element with empty value
if (!is_array($bodyParams) || empty($bodyParams)
    || (1 == count($bodyParams) && '' === reset($bodyParams))
) {
```

so in case when `$content` has value`name=` (or just `some string value`) the above condition is satisfied.

In apigility, for both these cases, removed condition was not satisfied (because `$bodyParams[0]` was not set). Only one issue was there when content was `0=`, `0=something` (or just `0`). In all of these cases it was *correctly* parsed to an array (with key `0`), but because of the (removed) condition, `$bodyParams` has then string raw content (`0=`, `0=something`, `0` respectively). In this case fatal error occurs due to type mismatch (`setBodyParams()` requires an array). Apigility is built to have always body parameters as an array and it's not allowing to have string value (unlike the `AbstractRestfulController`).

**To emphasize again:** I can't find any example when `!is_array($bodyParams)` condition will be satisfied.

In this situation, I've decided to remove the condition at all, because as long as `$bodyParams` is always an array, there is no any issue. I think, this is better approach, because probably we don't want to return any `ApiProblemResponse` in case when content is `name=` (please notice, in this situation original condition in `AbstractRestfulController` will be satisfied!).

In conclusion, as we know that `$bodyParams` is always array, line 120 was also redundant, so it's removed.

### CODE REVIEW

**Comment by weierophinney** [comment]
Thanks, @webimpress
